### PR TITLE
:sparkles: creation of !lastlane command

### PR DESCRIPTION
**Pull request by Arcyvilk**
You proud of me, @Aergwyn senpai?

### CODE REVIEW

File: lib/commands/commands/riot.ts [hunk 1]
```diff
@@ -32,39 +34,157 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+        players[i].summ1 = participant.spell1Id;
+        players[i].summ2 = participant.spell2Id;
+        players[i].fbkill = participant.stats.firstBloodKill;
+        players[i].fbassist = participant.stats.firstBloodAssist;
+        players[i].role = participant.timeline.role;
+        players[i].lane = participant.timeline.lane;
+    })
+    const ourPlayer = players.find((player:any) => player.id === playerId);
+    const enemies = players.filter((player:any) => player.lane === ourPlayer.lane && player.teamId !== ourPlayer.teamId);
+    const lane = cache["lanes"].find((lane:any) => lane.lane == ourPlayer.lane)
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]: What is this cache exactly?

File: lib/commands/commands/riot.ts [hunk 2]
```diff
@@ -32,39 +34,157 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+    const queue = cache["queues"].find((queue:any) => queue.id == lastGameData.data.queueId)
+
+    const embed = new Discord.RichEmbed()
+        .setColor('FDC000')
+        .setThumbnail(lane ? lane.icon : 'https://d1nhio0ox7pgb.cloudfront.net/_img/g_collection_png/standard/256x256/question.png')
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]: Could use a Discord one instead, less likely to randomly disappear.

File: lib/commands/commands/riot.ts [hunk 3]
```diff
@@ -32,39 +34,157 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+                        id: enemy.gameId,
+                        cs: player.minionsKilled + player.jungleMinionsKilled,
+                        gold: player.totalGold,
+                        level: player.level,
+                    })
+                })
+            }
+        }
+
+        const min5 = gameFrames.min5
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]: Can't this whole block be simplified? I see a lot of repetition and the conditions are really hard to read.

File: lib/commands/commands/riot.ts (hunk 2 above)
Comment (Arcyvilk) [reply lib/commands/commands/riot.ts]: Does Discord have some kind of placeholder picture free to use?

Comment (Aergwyn) [reply lib/commands/commands/riot.ts]: Oh it's the big image in the top-right corner? I misunderstood in that case.

File: lib/commands/commands/riot.ts (hunk 1 above)
Comment (Arcyvilk) [reply lib/commands/commands/riot.ts]: At the bot's startup, bot loads the static data from my database into this cache. I didn't have time to document it anywhere yet. In majority of cases it's just copy of static data from Riot's API.
In this case, `lanes` associates the lane's name with the corresponding icon to be displayed in the command 
image

File: lib/commands/commands/riot.ts (hunk 3 above)
Comment (Arcyvilk) [reply lib/commands/commands/riot.ts]: Fixed in f81071e

**Comment by Arcyvilk** [comment]
Current command's look:
image

File: lib/commands/commands/riot.ts [hunk 4]
```diff
@@ -32,39 +34,149 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+    else {
+        const relevantMinutes = [ 5, 10, 15 ];
+        const gameFrames:any = { };
+
+        embed
+            .setTitle(`${ourPlayer.champion 
+                ? ourPlayer.champion.name.toUpperCase() 
+                : '???'
+            } vs${enemies.map((enemy:any) => enemy.champion ? ` ${enemy.champion.name.toUpperCase()}` : '???')}`)
+            .setDescription(`**${ourPlayer.name}'s** ${ourPlayer.champion ? ourPlayer.champion.name : '???'
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]: > {ourPlayer.champion ? ourPlayer.champion.name : '???'

This still repeats everywhere. It has to go.

File: lib/commands/commands/riot.ts [hunk 5]
```diff
@@ -32,39 +34,149 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+                Object.values(timeline).map((player:any) => {
+                    const currentPlayer = players.find((p:any) => p.gameId === player.participantId);
+                    if (currentPlayer && currentPlayer.teamId !== ourPlayer.teamId && currentPlayer.lane !== ourPlayer.lane)
+                        gameFrames[`min${minute}`].enemyTeam += player.totalGold;
+                    if (currentPlayer && currentPlayer.teamId === ourPlayer.teamId && currentPlayer.lane !== ourPlayer.lane)
+                        gameFrames[`min${minute}`].allyTeam += player.totalGold;
+                });
+
+                const playerChampionName = ourPlayer.champion ? ourPlayer.champion.name : '[unknown ally champion]';
+                const enemyChampionName = enemies[0].champion ? enemies[0].champion.name : '[unknown senemy champion]';
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]:
```suggestion
                const enemyChampionName = enemies[0].champion ? enemies[0].champion.name : '[unknown enemy champion]';
```

File: lib/commands/commands/riot.ts [hunk 6]
```diff
@@ -32,39 +34,149 @@ export const getSummonerId = async (ign:string, server:string) => {
...
+                const playerGold = gameFrames[`min${minute}`].player.gold;
+                const playerCs = gameFrames[`min${minute}`].player.cs;
+                const enemyCs = gameFrames[`min${minute}`].enemies[0].cs;
+                const enemyGold = gameFrames[`min${minute}`].enemies[0].gold;
+                const allyTeamGoldAdvantage = gameFrames[`min${minute}`].allyTeam - gameFrames[`min${minute}`].enemyTeam;
+                const minuteSummary = gameFrames[`min${minute}`]
+                    ? `${playerChampionName} has **${Math.abs(playerGold - enemyGold)}** gold ${playerGold - enemyGold > 0 ? 'advantage' : 'disadvantage'}.
+                        CS scores are ${playerChampionName}'s **${playerCs}** to ${enemyChampionName}'s **${enemyCs}**.
```
Comment (Aergwyn) [inline lib/commands/commands/riot.ts]: Before you were using `\n` and here it's not.

Comment (Arcyvilk) [reply lib/commands/commands/riot.ts]: I'm not using `\n` here because the backticks preserve whitespaces

**Pull request merged**